### PR TITLE
feat: Request.postDataJSON

### PR DESCRIPTION
**Pull request by mxschmitt**
This PR implements the `Request.postDataJSON` method as proposed in #1686.

Currently it handles `JSON` and `form-urlencoded` values which is probably enough for most use cases. Also thought about renaming it to e.g. `parsedPostData`.

### CODE REVIEW

File: src/network.ts [hunk 1]
```diff
@@ -153,7 +154,29 @@ export class Request {
...
+
+    if (contentType === 'application/x-www-form-urlencoded') {
+      const entries: Record<string, string> = {};
+      const parsed = new URLSearchParams(this._postData);
+      for (const [k, v] of parsed.entries())
+        entries[k] = v;
+      return entries;
+    }
+
+    if (contentType.startsWith('application/json'))
```
Comment (pavelfeldman) [inline src/network.ts:173]: I wonder if we should `JSON.parse` unconditionally at this point. Users will post w/o proper mime type and would expect `postDataJSON` to work. It'll throw if data is a not a JSON string, which is Ok.

File: test/network.spec.js [hunk 2]
```diff
@@ -140,6 +140,43 @@ describe('Request.postData', function() {
...
+
+  it("should parse the data if content-type is application/x-www-form-urlencoded", async({page, server}) => {
+    await page.goto(server.EMPTY_PAGE);
+    server.setRoute('/post', (req, res) => res.end());
+    let request = null;
+    page.on('request', r => request = r);
+    await page.setContent(`<form method='POST' action='/post'><input type='text' name='foo' value='bar'><input type='number' name='baz' value='123'><input type='submit'></form>`);
+    await page.click('input[type=submit]');
+    expect(request).toBeTruthy();
+    expect(request.postDataJSON()).toEqual({"foo":"bar","baz":"123"});
```
Comment (pavelfeldman) [inline test/network.spec.js:171]: nit: we use single quotes.

**Review by pavelfeldman — COMMENTED** [review]
This looks great! One small question about JSON mime type check.

File: docs/api.md [hunk 3]
```diff
@@ -3538,6 +3539,9 @@ Whether this request is driving frame's navigation.
 #### request.postData()
 - returns: <?[string]> Request's post body, if any.
 
+#### request.postDataJSON()
+- returns: <?[Object]> Parsed request's body if the request is JSON or form-urlencoded if any.
```
Comment (pavelfeldman) [inline docs/api.md:3543]: Let's expand below on the logic - I missed the `form-urlencoded` bit in the docs when I was reading it for the first time.

File: test/network.spec.js (hunk 2 above)
Comment (mxschmitt) [reply test/network.spec.js]: Would be cool to share a style guide somewhere, Visual Studio Code e.g. adds spaces before curly brackets etc. but thats not common in this project.

File: docs/api.md (hunk 3 above)
Comment (mxschmitt) [reply docs/api.md]: Feel free to adjust the wording, not sure if it's clear now.

**Pull request merged**